sui/src/sui_commands.rs

```diff
@@ -29,53 +30,112 @@ use crate::config::{
 use crate::gateway::{EmbeddedGatewayConfig, GatewayType};
 use crate::keystore::{Keystore, KeystoreType, SuiKeystore};
 
+const SUI_DIR: &str = ".sui";
+const SUI_CONFIG_DIR: &str = "sui_config";
+const SUI_NETWORK_CONFIG: &str = "network.conf";
+
 #[derive(StructOpt)]
 #[structopt(rename_all = "kebab-case")]
 pub enum SuiCommand {
     /// Start sui network.
     #[structopt(name = "start")]
     Start {
-        #[structopt(long, default_value = "./network.conf")]
-        config: PathBuf,
+        #[structopt(long)]
+        config: Option<PathBuf>,
     },
     #[structopt(name = "genesis")]
     Genesis {
-        #[structopt(long, default_value = ".")]
-        working_dir: PathBuf,
         #[structopt(long)]
-        config: Option<PathBuf>,
+        working_dir: Option<PathBuf>,
+        #[structopt(short, long, help = "Forces overwriting existing configuration")]
+        force: bool,
     },
 }
 
+pub fn sui_config_dir() -> Result<PathBuf, anyhow::Error> {
+    let mut p = match dirs::home_dir() {
+        Some(v) => v,
+        None => return Err(anyhow!("Cannot obtain home directory path")),
+    };
+    p.push(SUI_DIR);
+    p.push(SUI_CONFIG_DIR);
+    Ok(p)
+}
+
 impl SuiCommand {
     pub async fn execute(&self) -> Result<(), anyhow::Error> {
```

Comment:
We if pass self in as value instead of ref, we can save ourselves from all the cloning later in the codebase, and I think it make more sense as SuiCommand is only run once and won't be reused again.

Reply:
Looks like it's an independent refactoring to me.